### ISSUE TITLE
The unit test core does not flowing symbolic links during searching for tests

### ISSUE THREAD

**Issue by EugenWiens** [platformio/platformio-core#4288]
### Configuration

**Operating system**:
Ubuntu 20.04

**PlatformIO Version** (`platformio --version`): 
PlatformIO Core, version 6.0.1

### Description of problem
The unit test core does not flowing symbolic links during searching for tests.

#### Steps to Reproduce
Project with following striucture
Project
```
  `- src
  `- lib
      `- module1
            `-test
                 `- test_one
  `- test
        `- module1 -> ./lib/module1/test
```

### Actual Results
the unit test core does't find any tests

### Expected Results
the unit test core should find the folder `test_one` 

### If problems with PlatformIO Build System:

**Comment by ivankravets** [comment]
Thanks for the report. Please re-test with `pio upgrade --dev`. 

Does it work now?

**Comment by EugenWiens** [comment]
yes with 6.0.2b1 the symbolic link works again. When we can see it in the production version?

**Comment by ivankravets** [comment]
We plan Core 6.0.2 for today/tomorrow.

**Comment by lemonkey** [comment]
@4303  Thanks for bringing this up! 💯
